Wikia/pandora
# Initial checkin for DIS-81 Tests for DIS-41 Display ThreadList

* Add additional tests for getting specific forums with threads
* Add methods to convert HalResponses for Threads, Posts, and PostRevisions
* Add SecureActivity for VIEW_FORUM (stubbed impl ATM)
* Refactor ForumResourceTest to use Hamcrest assertThat
* Introduce new generic Hamcrest Matcher for Content comparisons by id from a list
* Refactor all Mappers to not stringify Integer types (we are only stringifying Long)
* Remove unnecessary ForumService dependency in the ThreadResource and add TODO to check forum existence when we implement create thread
* Add ThreadResource and ThreadService Test

TODO: Implement ThreadResource Integration test

Minor: checkstyle fixes in the separate commit (3c3e8c2)

@jimmordino @armonr
